tikv-client/src/main/java/com/pingcap/tikv/txn/LockResolverClientV4.java

```diff
@@ -342,17 +366,235 @@ private TxnStatus getTxnStatus(
         throw new KeyException(keyError);
       }
 
-      if (resp.getLockTtl() != 0) {
-        status = new TxnStatus(resp.getLockTtl(), 0L, resp.getAction());
+      status = new TxnStatus();
+      status.setAction(resp.getAction());
+      status.setPrimaryLock(resp.getLockInfo());
+
+      if (status.getPrimaryLock() != null && status.getPrimaryLock().getUseAsyncCommit()) {
+        if (!TsoUtils.isExpired(txnID, resp.getLockTtl())) {
+          status.setTtl(resp.getLockTtl());
+        }
+      } else if (resp.getLockTtl() != 0) {
+        status.setTtl(resp.getLockTtl());
       } else {
-        status = new TxnStatus(0L, resp.getCommitVersion(), resp.getAction());
+        status.setCommitTS(resp.getCommitVersion());
```

Comment:
Does status need to be re-initialised here or is it ok to inherit the old values of other fields?

Reply:
see line 369 `status = new TxnStatus();`
status will be recreated in each round.